staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -352,7 +360,20 @@ func (s preparedGenericAPIServer) Run(stopCh <-chan struct{}) error {
 	}()
 
 	// close socket after delayed stopCh
-	stoppedCh, listenerStoppedCh, err := s.NonBlockingRun(delayedStopCh.Signaled())
+	drainedCh := s.lifecycleSignals.InFlightRequestsDrained
+	stopHttpServerCh := delayedStopCh.Signaled()
+	shutdownTimeout := s.ShutdownTimeout
+	if s.KeepListeningDuringGracefulTermination {
+		// when this mode is enabled, we do the following:
+		// - the server will continue to listen until all existing
+		//   requests in flight have been drained
+		// - once drained, http Server Shutdown is invoked with a timeout of 5s
```

Comment:
what do the 5s mean? Why not immediately or 1s?

Reply:
I set it to `2s`, net/http waits for `1s` after sending GO_AWAY to take into account round trip time for the peer, so we should respect that.

Reply:
please add this in the comment. 5s seems to be arbitrary without explanation. Also sync comment and code :)